snuba/subscriptions/data.py

```diff
@@ -166,10 +180,71 @@ class SnQLSubscriptionData(SubscriptionData):
     def build_request(
         self, dataset: Dataset, timestamp: datetime, offset: Optional[int], timer: Timer
     ) -> Request:
-        # TODO: Parse the query without doing the full validation
-        # TODO: Add time range conditions
-        # TODO: add offset condition
-        raise NotImplementedError
+        schema = RequestSchema.build_with_extensions(
+            {}, SubscriptionRequestSettings, Language.SNQL,
+        )
+
+        def add_conditions(query: Union[CompositeQuery[Entity], Query]) -> None:
+            # TODO: Support composite queries with multiple entities.
```

Comment:
@evanh, when we will support joins we will need a better way to decide which entities we want to add timestamp conditions on. 
This is now just reflecting the same logic we have for legacy queries.

Reply:
What were you thinking of? I assumed we would pull all the entities referenced from the Join clause, and then add a condition for each entity. That's what the validator does when it checks for the conditions.

Reply:
> I assumed we would pull all the entities referenced from the Join clause, and then add a condition for each entity.

There are entities that do not have a time column at all (like the cdc ones).
Others may require time conditions that could be not the same between entities (example when we had spans. WThe time condition on spans would have been larger than transactions as the transaction timestamp represent the end of the transaciton).

The second case is not a thing nowdays. For the first one I think we may just want to use the required timestamp column field you use in the validators. Though I am not sure if "required timestamp condition" should automatically mean "time range condition". Opinions ?

Reply:
That makes sense to me.